Makefile

```diff
@@ -1,17 +1,46 @@
-.PHONY: build run default
+.PHONY: dg doc build run test clean
 
-IMAGE_NAME = dovecot
+DISTRO = fedora-26-x86_64
+VARIANT = upstream
+DG = /usr/bin/dg
+GOMD2MAN = /usr/bin/go-md2man
+DOCKERFILE_SRC := Dockerfile.template
+DOCKERFILE := Dockerfile
+TEST_IMAGE_NAME := container-images-tests
 
+SELECTORS = --distro ${DISTRO}.yaml --multispec-selector variant=${VARIANT}
+DG_EXEC = ${DG} --max-passes 25 --spec specs/configuration.yml --multispec specs/multispec.yml ${SELECTORS}
+DISTRO_ID = $(shell ${DG_EXEC} --template "{{ config.os.id }}")
+IMAGE_REPOSITORY = $(shell ${DG_EXEC} --template "{{ spec.image_repository }}")
 
-default: run
+install-dependencies:
+	./requirements.sh
 
-build:
-	docker build --tag=$(IMAGE_NAME) .
+dg:
+	${DG_EXEC} --template $(DOCKERFILE_SRC) --output $(DOCKERFILE)
+	${DG_EXEC} --template help/help.md --output help/help.md.rendered
+
+doc: dg
+	mkdir -p ./root/
+	${GOMD2MAN} -in=help/help.md.rendered -out=./root/help.1
+
+build: doc dg
+	sudo chown -R $(shell id -u) .
```

Comment:
I'm wondering if it'd make more sense to to create the rendered/generated files by using redirection, e.g. `command > Dockerfile` - that way we'd have resulting files owned by current user and would avoid the `sudo` call - does that make sense?

Reply:
I did not know, it is possible